src/mlpack/methods/matrix_completion/matrix_completion.hpp

```diff
@@ -13,50 +14,65 @@
 #ifndef MLPACK_METHODS_MATRIX_COMPLETION_MATRIX_COMPLETION_HPP
 #define MLPACK_METHODS_MATRIX_COMPLETION_MATRIX_COMPLETION_HPP
 
-#include <mlpack/core/optimizers/sdp/sdp.hpp>
-#include <mlpack/core/optimizers/sdp/lrsdp.hpp>
+#include "mc_sdp_solver.hpp"
+#include "mc_fw_solver.hpp"
 
 namespace mlpack {
 namespace matrix_completion {
 
 /**
- * This class implements the popular nuclear norm minimization heuristic for
- * matrix completion problems. That is, given known values M_ij's, the
- * following optimization problem (semi-definite program) is solved to fill in
- * the remaining unknown values of X
+ * This class implements a wrapper for different types of solvers of matrix
+ * completion problems. That is, we want to recover a large low rank matrix,
+ * with only a few elements in the matrix known. There are quite many
+ * mathematical models and numerical algorithms to solve this kind of problems.
+ * This class only provide a wrapper using class templates.
  *
- *   min ||X||_* subj to X_ij = M_ij
+ * The model solved using SDP solver is:
+ *
+ * \f[
+ *   min ||X||_* subj to X_ij = M_ij, \qquad (i,j) \in \Omega
+ * \f]
  *
  * where ||X||_* denotes the nuclear norm (sum of singular values of X).
  *
- * For a theoretical treatment of the conditions necessary for exact recovery,
- * see the following paper:
+ * For a theoretical treatment of the conditions necessary for exact recovery
+ * using the above model, see the following paper:
  *
  *   A Simpler Appoarch to Matrix Completion.
```

Comment:
Do you mind to use bibtex here? (example: https://github.com/mlpack/mlpack/blob/master/src/mlpack/core/optimizers/parallel_sgd/parallel_sgd.hpp#L27)